### PR TITLE
[BE][accelerator] formalize API name `{current,set}_device_{idx => index}`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140542

### CODE REVIEW

**Review by guangyey — APPROVED** [review]

**Comment by guangyey** [comment]
Good to me, it needs Alban's final stamp.

**Comment by XuehaiPan** [comment]
@albanD could you take a look at this?

File: docs/source/conf.py [hunk 1]
```diff
@@ -158,6 +158,9 @@
     "get_model_info",
     "get_storage_info",
     "hierarchical_pickle",
+    # torch.accelerator
```
Comment (albanD) [inline docs/source/conf.py:161]: Adding entries to this and the allowlist below is not ok. We need to keep these functions documented until they are removed.

**Review by albanD — COMMENTED** [review]
Both names seemed used so no strong opinion here. I guess these are new enough that we can change them.

File: docs/source/conf.py (hunk 1 above)
Comment (XuehaiPan) [reply docs/source/conf.py:161]: Updated.

**Review by albanD — APPROVED** [review]
SGTM !

**Comment by guangyey** [comment]
@XuehaiPan Could you help clean up {idx => index} in https://github.com/pytorch/pytorch/blob/main/test/test_accelerator.py

**Comment by guangyey** [comment]
@albanD Do you think if this PR should cherry pick to release/2.6

**Pull request closed without merging**